src/test/java/rx/internal/operators/OnSubscribeFromArrayTest.java

```diff
@@ -64,4 +65,15 @@ public void backpressure() {
         ts.assertCompleted();
     }
 
+    @Test
+    public void empty() {
+        Assert.assertSame(Observable.empty(), Observable.from(new Object[0]));
+    }
+
+    @Test
+    public void just() {
+        Observable<Integer> source = Observable.from(new Integer[] { 1 });
+        Assert.assertTrue(source.getClass().toString(), source instanceof ScalarSynchronousObservable);
```

Comment:
I don't like these two tests. It's the kind code coverage erroneously lures you in to. We're testing that an implementation detail is true. I would prefer a test the just asserted 0, 1, and N elements behaves as expected and leave it up to the perf tests to ensure there's no regression in performance on changes.